langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -18,73 +18,229 @@
+        if (value1 == null) {
+            if (value2 == null) {
+                return 0;
+            }
+            if (isAscending) {
+                return 1;
+            }
+            return -1;
+        }
+        if (value2 == null) {
+            if (isAscending) {
+                return -1;
+            }
+            return 1;
+        }
+        if (TypeTags.isIntegerTypeTag(type.getTag())) {
+            return Long.compare((long) value1, (long) value2);
+        } else if (type.getTag() == TypeTags.FLOAT_TAG) {
+            // NaN should be placed last or one before the last when () is present irrespective of the sort direction.
+            if (Double.isNaN((double) value1)) {
+                if (Double.isNaN((double) value2)) {
+                    return 0;
+                }
+                if (isAscending) {
+                    return 1;
+                }
+                return -1;
+            }
+            if (Double.isNaN((double) value2)) {
+                if (isAscending) {
+                    return -1;
+                }
+                return 1;
+            }
+            // -0.0 = +0.0
+            if ((double) value1 == 0 && (double) value2 == 0) {
+                return 0;
+            }
+            return Double.compare((double) value1, (double) value2);
+        } else if (type.getTag() == TypeTags.DECIMAL_TAG) {
+            return new BigDecimal(value1.toString()).compareTo(new BigDecimal(value2.toString()));
+        } else if (type.getTag() == TypeTags.BOOLEAN_TAG) {
+            return Boolean.compare((boolean) value1, (boolean) value2);
+        } else if (TypeTags.isStringTypeTag(type.getTag())) {
+            return codePointCompare(value1.toString(), value2.toString());
+        } else if (type.getTag() == TypeTags.BYTE_TAG) {
+            return Integer.compare((int) value1, (int) value2);
+        } else if (type.getTag() == TypeTags.ARRAY_TAG) {
+            int lengthVal1 = ((ArrayValue) value1).size();
+            int lengthVal2 = ((ArrayValue) value2).size();
+            if (lengthVal1 == 0) {
+                if (lengthVal2 == 0) {
+                    return 0;
+                }
+                return -1;
+            }
+            if (lengthVal2 == 0) {
+                return 1;
+            }
+            int len = Math.min(lengthVal1, lengthVal2);
+            int c = 0;
+            for (int i = 0; i < len; i++) {
+                c = sortFunc(((ArrayValue) value1).get(i), ((ArrayValue) value2).get(i),
+                        ((BArrayType) type).getElementType(), isAscending);
+                if (c != 0) {
+                    break;
+                } else {
+                    if (i == len - 1 && lengthVal1 < lengthVal2) {
+                        return -1;
+                    }
+                }
+            }
+            return c;
+        }
+        throw BallerinaErrors.createError("Invalid type to sort", "Expected an ordered type, but found type: "
```

Comment:
This error message needs to conform to the same format we follow for other lang lib errors. You can check the other places in the lang lib where panics are created for reference. Also, IIRC, we don't start the detail with a capital letter. @MaryamZi